src/SharpGLTF.Core/Schema2/gltf.MeshPrimitive.CESIUM_primitive_outline.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SharpGLTF.Schema2
+{
+    partial class CESIUM_primitive_outlineglTFprimitiveextension
+    {
+        internal CESIUM_primitive_outlineglTFprimitiveextension(MeshPrimitive meshPrimitive) { }
+
+        public int? Indices
+        {
+            get => _indices;
+            set => _indices = value;
+        }
+    }
+    partial class MeshPrimitive
+    {
+        public void SetCesiumOutline(Accessor accessor)
+        {
+            if (accessor == null) { RemoveExtensions<CESIUM_primitive_outlineglTFprimitiveextension>(); return; }
+
+            Guard.NotNull(accessor, nameof(accessor));
+            Guard.MustShareLogicalParent(LogicalParent.LogicalParent, "this", accessor, nameof(accessor));
+            Guard.IsTrue(accessor.Encoding == EncodingType.UNSIGNED_INT, nameof(accessor));
+            Guard.IsTrue(accessor.Dimensions == DimensionType.SCALAR, nameof(accessor));
+
+            var ext = UseExtension<CESIUM_primitive_outlineglTFprimitiveextension>();
+            ext.Indices = accessor.LogicalIndex;
+        }
+    }
+
+    public static class CesiumToolkit
+    {
+        /// <summary>
+        /// Creates an accessor to store Cesium outline vertex indices
+        /// </summary>
+        /// <param name="model"></param>
+        /// <param name="outlines"></param>
+        /// <returns></returns>
+        public static Accessor CreateCesiumOutlineAccessor(ModelRoot model, IReadOnlyList<uint> outlines)
+        {
+            var outlineBytes = new List<byte>();
+
+            foreach (var outline in outlines)
+            {
+                var bytes = BitConverter.GetBytes(outline).ToList();
+                outlineBytes.AddRange(bytes);
+            }
+            var buffer = model.UseBufferView(outlineBytes.ToArray());
+            var accessor = model.CreateAccessor("Cesium outlines");
+            accessor.SetData(buffer, 0, outlineBytes.Count / 4, DimensionType.SCALAR, EncodingType.UNSIGNED_INT, false);
+            return accessor;
+        }
+
+        /// <summary>
+        /// Checks if all the indices in the Cesium Outline Extension are existing in the primitive indices 
+        /// </summary>
+        /// <param name="model"></param>
+        /// <returns>boolean, true is valid false is not valid</returns>
+        public static bool ValidateCesiumOutlineExtension(ModelRoot model)
```

Comment:
This is a method that checks the indices of the used Cesium outline accessor against the mesh primitive indices. Todo: call this method from the glTF validator (if I know where to do this).